<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <!-- jQuery -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.10.3/jquery-ui.min.js"></script>
  
  <!-- bootstrap -->
  <!--<link href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" rel="stylesheet"  id="style">-->
  <script src="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/js/bootstrap.min.js"></script>
  
  <!-- highlight.js -->
  <!--<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/default.min.css" rel="stylesheet" id="code-style">-->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/highlight.min.js"></script>
  <script>
  hljs.LANGUAGES.r=function(a){var b="([a-zA-Z]|\\.[a-zA-Z.])[a-zA-Z0-9._]*";return{c:[a.HCM,{b:b,l:b,k:{keyword:"function if in break next repeat else for return switch while try tryCatch|10 stop warning require library attach detach source setMethod setGeneric setGroupGeneric setClass ...|10",literal:"NULL NA TRUE FALSE T F Inf NaN NA_integer_|10 NA_real_|10 NA_character_|10 NA_complex_|10"},r:0},{cN:"number",b:"0[xX][0-9a-fA-F]+[Li]?\\b",r:0},{cN:"number",b:"\\d+(?:[eE][+\\-]?\\d*)?L\\b",r:0},{cN:"number",b:"\\d+\\.(?!\\d)(?:i\\b)?",r:0},{cN:"number",b:"\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[a.BE],r:0},{cN:"string",b:"'",e:"'",c:[a.BE],r:0}]}}(hljs); </script>
  <!--<script type="text/javascript", src="https://yandex.st/highlightjs/7.3/languages/r.min.js"></script>-->
  
  <!-- Manific Popup -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/magnific-popup.js/0.8.9/jquery.magnific-popup.min.js"></script>
  
  <script type="text/javascript"
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script>
  <script defer="defer">
  // Function to generate the dynamic table of contents
  jQuery.fn.generate_TOC = function () {
    var base = $(this[0]);
  
    var selectors = ['h1', 'h2', 'h3', 'h4'];
  
    var last_ptr = [{}, {}, {}, {}];
  
    var anchors = {};
  
    generate_anchor = function (text) {
      var test = text.replace(/\W/g, '_');
  
      while(test in anchors){
        //if no suffix, add one
        if(test.match(/_\d+$/) === null){
          test = test + "_2";
        }
        //else generate unique id for duplicates by adding one to the suffix
        else {
          test = test.replace(/_(\d+)$/, function(match, number){ var num=+number+1; return("_" + num) });
        }
      }
      anchors[test]=1;
      return(test);
    }
  
    $(selectors.join(',')).filter(function(index) { return $(this).parent().attr("id") != 'header'; }).each(function () {
  
      var heading = $(this);
      var idx = selectors.indexOf(heading.prop('tagName').toLowerCase());
      var itr = 0;
  
      while (itr <= idx) {
        if (jQuery.isEmptyObject(last_ptr[itr])) {
          last_ptr[itr] = $('<ul>').addClass('nav');
          if (itr === 0) {
            base.append(last_ptr[itr])
          } else {
            if(last_ptr[itr-1].children('li').length === 0){
              last_ptr[itr-1].append(last_ptr[itr]);
            }
            else {
              last_ptr[itr - 1].children('li').last().append(last_ptr[itr]);
            }
          }
        }
        itr++;
      }
      var anchor = generate_anchor(heading.text());
      heading.attr('id', anchor);
      var a = $('<a>')
      .text(heading.text())
      .attr('href', '#' + anchor);
  
    var li = $('<li>')
      .append(a);
  
    last_ptr[idx].append(li);
    for (i = idx + 1; i < last_ptr.length; i++) {
      last_ptr[i] = {};
    }
    });
  }
  /* run scripts when document is ready */
  $(function() {
    "use strict";
  
    var $window = $(window);
    var $body = $(document.body);
  
    /* size of thumbnails */
  
    var hidden_types = ['source']
    var output_types = ['output', 'message', 'warning', 'error']
  
    /* style tables */
    $('table').addClass('table table-striped table-bordered table-hover table-condensed');
  
    $('pre code').each(function(i, e) {
      hljs.highlightBlock(e);
    });
  
    /* Magnific Popup */
    $(".thumbnail").each(function(){
      $(this).magnificPopup({
        disableOn: 768,
        closeOnContentClick: true,
  
        type: 'image',
        items: {
          src: $(this).find('img').attr('src'),
        }
      });
    });
  
    function toggle_block(obj, show) {
      var span = obj.find('span');
      if(show === true){
        span.removeClass('glyphicon-chevron-up').addClass('glyphicon-chevron-down');
        obj.next('pre').slideDown();
      }
      else {
        span.removeClass('glyphicon-chevron-down').addClass('glyphicon-chevron-up');
        obj.next('pre').slideUp();
      }
    }
  
    function toggle_thumbnails(imgs, show){
      if(show === true){
        imgs.parents().show()
        imgs.slideDown();
      }
      else {
        imgs.slideUp(400, function(){ $(this).parent().hide(); });
      }
    }
  
    function global_toggle(obj){
      var type = obj.attr('type');
      var show = !obj.parent('li').hasClass('active');
      if(show === true){
        obj.parent('li').addClass('active');
      }
      else{
        obj.parent('li').removeClass('active');
      }
      if(type == 'figure'){
        toggle_thumbnails($('.thumbnail img'), show);
      }
      else {
        $('.toggle.' + type).each(function() { toggle_block($(this), show); });
      }
    }
  
    /* onclick toggle next code block */
    $('.toggle').click(function() {
      var span = $(this).find('span');
      toggle_block($(this), !span.hasClass('glyphicon-chevron-down'));
      return false
    })
  
    // global toggles
    $('.toggle-global').click(function(){
      var type = $(this).attr('type');
      if(type === 'all-source'){
          $('li a.source').each(function() {
            global_toggle($(this));
          });
        }
      else if(type === 'all-output'){
        $.each(output_types, function(i, val){
          console.log(val);
          global_toggle($('li a.' + val));
        });
      }
      else {
        console.log($(this));
        global_toggle($(this));
      }
      return false;
    });
    /* table of contents */
    if($(['h1', 'h2', 'h3', 'h4'].join(',')).length > 0){
      $('body > #wrap > .container > .row').append('<div class="col-md-2"><div id="toc" class="well sidebar sidenav affix hidden-print"/></div>');
      $('#toc').generate_TOC();
    }
  
    $.each(hidden_types, function(i, type) {
      $('li[type=' + type + ']').each(function(){ global_toggle($(this)); });
    });
  
    /* remove paragraphs with no content */
    $('p:empty').remove();
  
    $body.scrollspy({
      target: '.sidebar',
    });
  
    /* theme switch */
    $('.theme-switch').click(function(){
      var css = $('link[title=' + $(this).attr('title') + ']');
      $('#theme[rel=stylesheet]').attr('href', css.attr('href'));
      $('.theme-switch').closest('li').removeClass('active');
      $(this).closest('li').addClass('active');
      return false;
    });
    /* code style switch */ //TODO use same function for both of these?
    $('.highlight-switch').click(function(){
      var css = $('link[title="' + $(this).attr('title') + '"]');
      $('#highlight[rel=stylesheet]').attr('href', css.attr('href'));
      $('.highlight-switch').closest('li').removeClass('active');
      $(this).closest('li').addClass('active');
      return false;
    });
  
    //TODO refresh on show/hide
    $window.on('load', function () {
      $body.scrollspy('refresh');
    })
  
  });
  
  </script>
  <style>
  /* Knitr_bootstrap styles */
  #header {
    display: none !important;
    visibility: hidden !important;
  }
  #wrap .container-fluid {
    padding: 0;
    overflow: hidden;
  }
  .toggle{
    text-transform: capitalize;
  }
  
  .toggle-global{
    text-transform: capitalize;
  }
  
  /* Sticky footer styles */
  * {
    margin:0;
  }
  html,
  body {
      height: 100%;
      padding:0 !important;
      /* The html and body elements cannot have any padding or margin. */
      /*overflow-x: hidden;*/
  }
  
  /* Wrapper for page content to push down footer */
  #wrap {
      min-height: 100%;
      height: auto !important;
      height: 100%;
      /* Negative indent footer by it's height */
      margin: 0 auto -120px;
  }
  
  /* Set the fixed height of the footer here */
  #push,
  #footer {
      height: 120px;
  }
  
  #footer {
    text-align: center;
  }
  
  /* Top level subheader elements.  These are the first nested items underneath a header element. */
  .header li {
    font-size: 20px;
  }
  
  /* Makes the font smaller for all subheader elements. */
  .sub-header li {
      font-size: 12px;
  }
  
  button.thumbnails {
    margin-left:0px;
  }
  
  /*
   * Side navigation
   *
   * Scrollspy and affixed enhanced navigation to highlight sections and secondary
   * sections of docs content.
   */
  
  /* By default it's not affixed in mobile views, so undo that */
  .sidebar.affix {
    position: static;
  }
  
  /* First level of nav */
  .sidenav {
    margin-top: 30px;
    margin-bottom: 30px;
    padding-top:    10px;
    padding-bottom: 10px;
    border-radius: 5px;
  }
  
  /* All levels of nav */
  .sidebar .nav > li > a {
    display: block;
    padding: 5px 20px;
  }
  .sidebar .nav > li > a:hover,
  .sidebar .nav > li > a:focus {
    text-decoration: none;
    border-right: 1px solid;
  }
  .sidebar .nav > .active > a,
  .sidebar .nav > .active:hover > a,
  .sidebar .nav > .active:focus > a {
    font-weight: bold;
    background-color: transparent;
    border-right: 1px solid;
  }
  
  /* Nav: second level (shown on .active) */
  .sidebar .nav .nav {
    display: none; /* Hide by default, but at >768px, show it */
    margin-bottom: 8px;
  }
  .sidebar .nav .nav > li > a {
    padding-top:    3px;
    padding-bottom: 3px;
    padding-left: 30px;
    font-size: 90%;
  }
  
  .sidebar .nav .nav .nav > li > a {
    padding-left: 40px;
  }
  .sidebar .nav .nav .nav .nav > li > a {
    padding-left: 50px;
  }
  
  /* Show and affix the side nav when space allows it */
  @media screen and (min-width: 992px) {
    .sidebar .nav > .active > ul {
      display: block;
    }
    /* Widen the fixed sidebar */
    .sidebar.affix,
    .sidebar.affix-bottom {
      width: 213px;
    }
    .sidebar.affix-top,
    .sidebar.affix {
      position: fixed; /* Undo the static from mobile first approach */
      top: 30px;
    }
    .sidebar.affix-bottom {
      position: absolute; /* Undo the static from mobile first approach */
    }
    .sidebar.affix-bottom .sidenav,
    .sidebar.affix .sidenav {
      margin-top: 0;
      margin-bottom: 0;
    }
  }
  @media screen and (min-width: 1200px) {
    /* Widen the fixed sidebar again */
    .sidebar.affix-bottom,
    .sidebar.affix {
      width: 263px;
    }
  }
  
  #toc {
    padding: 10px 0px;
    margin:0;
    border:0;
  }
  
  
  .panel pre {
    margin: 0;
    padding: 0;
    border: 0;
  }
  button + pre {
    margin: 0;
    padding: 0;
  }
  pre code {
    border-radius: 0;
  }
  /* Magnific Popup CSS */
  .mfp-bg {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 1042;
    overflow: hidden;
    position: fixed;
    background: #0b0b0b;
    opacity: 0.8;
    filter: alpha(opacity=80); }
  
  .mfp-wrap {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 1043;
    position: fixed;
    outline: none !important;
    -webkit-backface-visibility: hidden; }
  
  .mfp-container {
    text-align: center;
    position: absolute;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    padding: 0 8px;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box; }
  
  .mfp-container:before {
    content: '';
    display: inline-block;
    height: 100%;
    vertical-align: middle; }
  
  .mfp-align-top .mfp-container:before {
    display: none; }
  
  .mfp-content {
    position: relative;
    display: inline-block;
    vertical-align: middle;
    margin: 0 auto;
    text-align: left;
    z-index: 1045; }
  
  .mfp-inline-holder .mfp-content,
  .mfp-ajax-holder .mfp-content {
    width: 100%;
    cursor: auto; }
  
  .mfp-ajax-cur {
    cursor: progress; }
  
  .mfp-zoom-out-cur,
  .mfp-zoom-out-cur .mfp-image-holder .mfp-close {
    cursor: -moz-zoom-out;
    cursor: -webkit-zoom-out;
    cursor: zoom-out; }
  
  .mfp-zoom {
    cursor: pointer;
    cursor: -webkit-zoom-in;
    cursor: -moz-zoom-in;
    cursor: zoom-in; }
  
  .mfp-auto-cursor .mfp-content {
    cursor: auto; }
  
  .mfp-close,
  .mfp-arrow,
  .mfp-preloader,
  .mfp-counter {
    -webkit-user-select: none;
    -moz-user-select: none;
    user-select: none; }
  
  .mfp-loading.mfp-figure {
    display: none; }
  
  .mfp-hide {
    display: none !important; }
  
  .mfp-preloader {
    color: #cccccc;
    position: absolute;
    top: 50%;
    width: auto;
    text-align: center;
    margin-top: -0.8em;
    left: 8px;
    right: 8px;
    z-index: 1044; }
  
  .mfp-preloader a {
    color: #cccccc; }
  
  .mfp-preloader a:hover {
    color: white; }
  
  .mfp-s-ready .mfp-preloader {
    display: none; }
  
  .mfp-s-error .mfp-content {
    display: none; }
  
  button.mfp-close,
  button.mfp-arrow {
    overflow: visible;
    cursor: pointer;
    background: transparent;
    border: 0;
    -webkit-appearance: none;
    display: block;
    padding: 0;
    z-index: 1046;
    -webkit-box-shadow: none;
    box-shadow: none; }
  
  button::-moz-focus-inner {
    padding: 0;
    border: 0; }
  
  .mfp-close {
    width: 44px;
    height: 44px;
    line-height: 44px;
    position: absolute;
    right: 0;
    top: 0;
    text-decoration: none;
    text-align: center;
    opacity: 0.65;
    padding: 0 0 18px 10px;
    color: white;
    font-style: normal;
    font-size: 28px;
    font-family: Arial, Baskerville, monospace; }
    .mfp-close:hover, .mfp-close:focus {
      opacity: 1; }
    .mfp-close:active {
      top: 1px; }
  
  .mfp-close-btn-in .mfp-close {
    color: #333333; }
  
  .mfp-image-holder .mfp-close,
  .mfp-iframe-holder .mfp-close {
    color: white;
    right: -6px;
    text-align: right;
    padding-right: 6px;
    width: 100%; }
  
  .mfp-counter {
    position: absolute;
    top: 0;
    right: 0;
    color: #cccccc;
    font-size: 12px;
    line-height: 18px; }
  
  .mfp-arrow {
    position: absolute;
    opacity: 0.65;
    margin: 0;
    top: 50%;
    margin-top: -55px;
    padding: 0;
    width: 90px;
    height: 110px;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0); }
  
  .mfp-arrow:active {
    margin-top: -54px; }
  
  .mfp-arrow:hover,
  .mfp-arrow:focus {
    opacity: 1; }
  
  .mfp-arrow:before, .mfp-arrow:after,
  .mfp-arrow .mfp-b,
  .mfp-arrow .mfp-a {
    content: '';
    display: block;
    width: 0;
    height: 0;
    position: absolute;
    left: 0;
    top: 0;
    margin-top: 35px;
    margin-left: 35px;
    border: medium inset transparent; }
  .mfp-arrow:after,
  .mfp-arrow .mfp-a {
    border-top-width: 13px;
    border-bottom-width: 13px;
    top: 8px; }
  .mfp-arrow:before,
  .mfp-arrow .mfp-b {
    border-top-width: 21px;
    border-bottom-width: 21px; }
  
  .mfp-arrow-left {
    left: 0; }
    .mfp-arrow-left:after,
    .mfp-arrow-left .mfp-a {
      border-right: 17px solid white;
      margin-left: 31px; }
    .mfp-arrow-left:before,
    .mfp-arrow-left .mfp-b {
      margin-left: 25px;
      border-right: 27px solid #3f3f3f; }
  
  .mfp-arrow-right {
    right: 0; }
    .mfp-arrow-right:after,
    .mfp-arrow-right .mfp-a {
      border-left: 17px solid white;
      margin-left: 39px; }
    .mfp-arrow-right:before,
    .mfp-arrow-right .mfp-b {
      border-left: 27px solid #3f3f3f; }
  
  .mfp-iframe-holder {
    padding-top: 40px;
    padding-bottom: 40px; }
  
  .mfp-iframe-holder .mfp-content {
    line-height: 0;
    width: 100%;
    max-width: 900px; }
  
  .mfp-iframe-scaler {
    width: 100%;
    height: 0;
    overflow: hidden;
    padding-top: 56.25%; }
  
  .mfp-iframe-scaler iframe {
    position: absolute;
    display: block;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    box-shadow: 0 0 8px rgba(0, 0, 0, 0.6);
    background: black; }
  
  .mfp-iframe-holder .mfp-close {
    top: -40px; }
  
  /* Main image in popup */
  img.mfp-img {
    width: auto;
    max-width: 100%;
    height: auto;
    display: block;
    line-height: 0;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    padding: 40px 0 40px;
    margin: 0 auto; }
  
  /* The shadow behind the image */
  .mfp-figure:after {
    content: '';
    position: absolute;
    left: 0;
    top: 40px;
    bottom: 40px;
    display: block;
    right: 0;
    width: auto;
    height: auto;
    z-index: -1;
    box-shadow: 0 0 8px rgba(0, 0, 0, 0.6);
    background: #444444; }
  
  .mfp-figure {
    line-height: 0; }
  
  .mfp-bottom-bar {
    margin-top: -36px;
    position: absolute;
    top: 100%;
    left: 0;
    width: 100%;
    cursor: auto; }
  
  .mfp-title {
    text-align: left;
    line-height: 18px;
    color: #f3f3f3;
    word-wrap: break-word;
    padding-right: 36px; }
  
  .mfp-figure small {
    color: #bdbdbd;
    display: block;
    font-size: 12px;
    line-height: 14px; }
  
  .mfp-image-holder .mfp-content {
    max-width: 100%; }
  
  .mfp-gallery .mfp-image-holder .mfp-figure {
    cursor: pointer; }
  
  @media screen and (max-width: 800px) and (orientation: landscape), screen and (max-height: 300px) {
    /**
     * Remove all paddings around the image on small screen
     */
    .mfp-img-mobile .mfp-image-holder {
      padding-left: 0;
      padding-right: 0; }
  
    .mfp-img-mobile img.mfp-img {
      padding: 0; }
  
    /* The shadow behind the image */
    .mfp-img-mobile .mfp-figure:after {
      top: 0;
      bottom: 0; }
  
    .mfp-img-mobile .mfp-bottom-bar {
      background: rgba(0, 0, 0, 0.6);
      bottom: 0;
      margin: 0;
      top: auto;
      padding: 3px 5px;
      position: fixed;
      -webkit-box-sizing: border-box;
      -moz-box-sizing: border-box;
      box-sizing: border-box; }
  
    .mfp-img-mobile .mfp-bottom-bar:empty {
      padding: 0; }
  
    .mfp-img-mobile .mfp-counter {
      right: 5px;
      top: 3px; }
  
    .mfp-img-mobile .mfp-close {
      top: 0;
      right: 0;
      width: 35px;
      height: 35px;
      line-height: 35px;
      background: rgba(0, 0, 0, 0.6);
      position: fixed;
      text-align: center;
      padding: 0; }
  
    .mfp-img-mobile .mfp-figure small {
      display: inline;
      margin-left: 5px; } }
  @media all and (max-width: 900px) {
    .mfp-arrow {
      -webkit-transform: scale(0.75);
      transform: scale(0.75); }
  
    .mfp-arrow-left {
      -webkit-transform-origin: 0;
      transform-origin: 0; }
  
    .mfp-arrow-right {
      -webkit-transform-origin: 100%;
      transform-origin: 100%; }
  
    .mfp-container {
      padding-left: 6px;
      padding-right: 6px; } }
  .mfp-ie7 .mfp-img {
    padding: 0; }
  .mfp-ie7 .mfp-bottom-bar {
    width: 600px;
    left: 50%;
    margin-left: -300px;
    margin-top: 5px;
    padding-bottom: 5px; }
  .mfp-ie7 .mfp-container {
    padding: 0; }
  .mfp-ie7 .mfp-content {
    padding-top: 44px; }
  .mfp-ie7 .mfp-close {
    top: 0;
    right: 0;
    padding-top: 0; }
  
  //Magnific overrides
  .mfp-image img{
    background: white;
  }
  .mfp-figure:after {
    background: white;
  }
  
  /*
   * Off Canvas navbar toggle right
   * --------------------------------------------------
   */
  
  @media screen and (max-width: 768px) {
    .row-offcanvas .collapsing {
    -webkit-transition: none 0;
      -moz-transition: none 0;
      transition: none 0;
    }
   .row-offcanvas .navbar {
    position: absolute;
    z-index: 2;
      right:0;
      height:100%;
      width:55px;
      border:0;
      background-color:transparent;
    }
    .row-offcanvas .navbar-toggle {
      margin-right: 5px;
      margin-left: 5px;
    }
    .row-offcanvas {
      position: relative;
    }
    .row-offcanvas-right.active .navbar {
    position: absolute;
    z-index: 2;
      right: -28.4%;
      width:40%;
      background-color:#eee;
      border:0 solid #ddd;
      border-left-width:1px;
    }
    .row-offcanvas-right.active {
      left: -30%;
    }
    .row-offcanvas-right.active .navbar-collapse {
      position: relative;
      width: 100%;
    }
    .row-offcanvas .content {
    /*width:calc(100% - 60px);*/
    }
  }
  </style>
</head>
<body>
<div id="wrap">
<div class="container">
<div class="row row-offcanvas row-offcanvas-right">
<div class="contents col-xs-12 col-md-10">
<h1>Gene counting from BigWig files</h1>
<p>This protocol was contributed by <a href="http://bit.ly/LColladoTorres">L. Collado-Torres</a> as is available at <a href="http://jtleek.com/protocols/">jtleek.com/protocols/</a>.</p>
<h1>Overview</h1>
<p>This short protocol shows how you can get a gene-level count matrix from BigWig files. We'll work with example data from <code>derfinderData</code> <a id='cite-derfinderData'></a>(<a href='https://github.com/lcolladotor/derfinderData'>Collado-Torres, Jaffe, and Leek, 2015</a>) and generate the count matrix using some custom code.</p>
<h1>Example</h1>
<p>First, lets locate a set of BigWig files from the amygdala.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Load data
library('derfinderData')

## Locate bigWig files
files <- dir(system.file('extdata', 'AMY', package = 'derfinderData'),  full.names = TRUE)
names(files) <- gsub('\\.bw', '', dir(system.file('extdata', 'AMY', package = 'derfinderData')))
head(files)</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##                                                                                                      HSB113 
## "/Library/Frameworks/R.framework/Versions/3.2patched/Resources/library/derfinderData/extdata/AMY/HSB113.bw" 
##                                                                                                      HSB123 
## "/Library/Frameworks/R.framework/Versions/3.2patched/Resources/library/derfinderData/extdata/AMY/HSB123.bw" 
##                                                                                                      HSB126 
## "/Library/Frameworks/R.framework/Versions/3.2patched/Resources/library/derfinderData/extdata/AMY/HSB126.bw" 
##                                                                                                      HSB130 
## "/Library/Frameworks/R.framework/Versions/3.2patched/Resources/library/derfinderData/extdata/AMY/HSB130.bw" 
##                                                                                                      HSB135 
## "/Library/Frameworks/R.framework/Versions/3.2patched/Resources/library/derfinderData/extdata/AMY/HSB135.bw" 
##                                                                                                      HSB136 
## "/Library/Frameworks/R.framework/Versions/3.2patched/Resources/library/derfinderData/extdata/AMY/HSB136.bw"
</code></pre>
</div>
<p>Next, lets define our set of genes.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">library('TxDb.Hsapiens.UCSC.hg19.knownGene')</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-up"></span> R message
</button>
<pre style="display:none"><code class="message r">## Loading required package: GenomicFeatures
## Loading required package: BiocGenerics
</code></pre>
<button class="warning R toggle btn btn-xs btn-warning">
<span class="glyphicon glyphicon-chevron-up"></span> R warning
</button>
<pre style="display:none"><code class="warning r">## Warning: package 'BiocGenerics' was built under R version 3.2.1
</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-up"></span> R message
</button>
<pre style="display:none"><code class="message r">## Loading required package: parallel
## 
## Attaching package: 'BiocGenerics'
## 
## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB
## 
## The following object is masked from 'package:stats':
## 
##     xtabs
## 
## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, as.vector, cbind,
##     colnames, do.call, duplicated, eval, evalq, Filter, Find, get,
##     grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
##     Position, rank, rbind, Reduce, rep.int, rownames, sapply,
##     setdiff, sort, table, tapply, union, unique, unlist, unsplit
## 
## Loading required package: S4Vectors
</code></pre>
<button class="warning R toggle btn btn-xs btn-warning">
<span class="glyphicon glyphicon-chevron-up"></span> R warning
</button>
<pre style="display:none"><code class="warning r">## Warning: package 'S4Vectors' was built under R version 3.2.1
</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-up"></span> R message
</button>
<pre style="display:none"><code class="message r">## Loading required package: stats4
## Loading required package: IRanges
</code></pre>
<button class="warning R toggle btn btn-xs btn-warning">
<span class="glyphicon glyphicon-chevron-up"></span> R warning
</button>
<pre style="display:none"><code class="warning r">## Warning: package 'IRanges' was built under R version 3.2.1
</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-up"></span> R message
</button>
<pre style="display:none"><code class="message r">## Loading required package: GenomeInfoDb
</code></pre>
<button class="warning R toggle btn btn-xs btn-warning">
<span class="glyphicon glyphicon-chevron-up"></span> R warning
</button>
<pre style="display:none"><code class="warning r">## Warning: package 'GenomeInfoDb' was built under R version 3.2.1
</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-up"></span> R message
</button>
<pre style="display:none"><code class="message r">## Loading required package: GenomicRanges
</code></pre>
<button class="warning R toggle btn btn-xs btn-warning">
<span class="glyphicon glyphicon-chevron-up"></span> R warning
</button>
<pre style="display:none"><code class="warning r">## Warning: package 'GenomicRanges' was built under R version 3.2.1
</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-up"></span> R message
</button>
<pre style="display:none"><code class="message r">## Loading required package: AnnotationDbi
</code></pre>
<button class="warning R toggle btn btn-xs btn-warning">
<span class="glyphicon glyphicon-chevron-up"></span> R warning
</button>
<pre style="display:none"><code class="warning r">## Warning: package 'AnnotationDbi' was built under R version 3.2.1
</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-up"></span> R message
</button>
<pre style="display:none"><code class="message r">## Loading required package: Biobase
## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## example data only has coverage for chr21, otherwise use all
txdb <- keepSeqlevels(TxDb.Hsapiens.UCSC.hg19.knownGene, 'chr21')

## Subset set of genes if you are not using all of them
ex <- exonsBy(txdb, by = 'gene')</code></pre>
</div>
<p>We can see that it's a GRangesList.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">class(ex)</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## [1] "GRangesList"
## attr(,"package")
## [1] "GenomicRanges"
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">ex</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## GRangesList object of length 296:
## $100129027 
## GRanges object with 3 ranges and 2 metadata columns:
##       seqnames               ranges strand |   exon_id   exon_name
##          <Rle>            <IRanges>  <Rle> | <integer> <character>
##   [1]    chr21 [47247755, 47251155]      - |    260437        <NA>
##   [2]    chr21 [47252522, 47252572]      - |    260438        <NA>
##   [3]    chr21 [47256200, 47256333]      - |    260439        <NA>
## 
## $100131902 
## GRanges object with 1 range and 2 metadata columns:
##       seqnames               ranges strand | exon_id exon_name
##   [1]    chr21 [31661463, 31661832]      - |  259463      <NA>
## 
## $100132288 
## GRanges object with 6 ranges and 2 metadata columns:
##       seqnames             ranges strand | exon_id exon_name
##   [1]    chr21 [9907189, 9907492]      - |  259089      <NA>
##   [2]    chr21 [9907189, 9908432]      - |  259090      <NA>
##   [3]    chr21 [9909047, 9909277]      - |  259091      <NA>
##   [4]    chr21 [9915250, 9916547]      - |  259092      <NA>
##   [5]    chr21 [9966322, 9966380]      - |  259094      <NA>
##   [6]    chr21 [9968516, 9968593]      - |  259095      <NA>
## 
## ...
## <293 more elements>
## -------
## seqinfo: 1 sequence from hg19 genome
</code></pre>
</div>
<p>We can use a similar approach to <a href="http://lcolladotor.github.io/protocols/bigwig_DEanalysis/">bigwig_DEanalysis</a> but it's a little bit more complicated because we are dealing with a GRangesList instead of a GRanges object.</p>
<p>First, we can create an exon level count matrix using <code>rtracklayer</code> <a id='cite-rtracklayer'></a>(<a href='http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract'>Lawrence, Gentleman, and Carey, 2009</a>).</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Import data and create exon count matrix
library('rtracklayer')</code></pre>
<button class="warning R toggle btn btn-xs btn-warning">
<span class="glyphicon glyphicon-chevron-up"></span> R warning
</button>
<pre style="display:none"><code class="warning r">## Warning: package 'rtracklayer' was built under R version 3.2.1
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">bw <- BigWigFileList(files)

exons <- unlist(ex)
counts_exons <- matrix(NA, nrow = length(exons), ncol = length(bw))
colnames(counts_exons) <- names(bw)
for(i in seq_len(length(bw))) {
    ## For more than one chr, you'll need to change this a bit
    coverage <- import(bw[[i]], as = 'RleList')$chr21
    counts_exons[, i] <- sum(Views(coverage, ranges(exons)))
    
    ## For more than one chr, you can use this code per chr or...
    ## something like this.
    # coverage <- import(bw[[i]], as = 'RleList')
    # for(chr in unique(seqnames(exons))) {
    #     counts_exons[seqnames(exons) == chr, i] <- sum(Views(coverage[[chr]], ranges(exons[seqnames(exons) == chr])))
    # }
}

## Divide by read length and round to integer numbers
counts_exons <- round(counts_exons / 76, 0)

## Lets explore the results
dim(counts_exons)</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## [1] 2822   12
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">head(counts_exons[, 1:6])</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##      HSB113 HSB123 HSB126 HSB130 HSB135 HSB136
## [1,]      0      0      0      0      0      0
## [2,]      0      0      0      0      0      0
## [3,]      0      0      0      0      0      0
## [4,]      0      0      0      0      0      0
## [5,]      0      1      3      1      3      1
## [6,]      0      3      7      5      6      3
</code></pre>
</div>
<p>We can then summary the exon count matrix into a gene count matrix.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Create gene count matrix
counts_genes <- matrix(NA, nrow = length(ex), ncol = length(bw))
colnames(counts_genes) <- names(bw)
for(i in seq_len(length(bw))) {
    counts_genes[, i]  <- sapply(split(counts_exons[, i], names(exons)), sum)
}

## Add gene names
rownames(counts_genes) <- names(ex)

## Explore the results
dim(counts_genes)</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## [1] 296  12
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">head(counts_genes[, 1:6])</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##           HSB113 HSB123 HSB126 HSB130 HSB135 HSB136
## 100129027      0      0      0      0      0      0
## 100131902      0      0      0      0      0      0
## 100132288      0      8     16      9     21      5
## 100133286      5     11     16     27      7     21
## 100151643      0      0      0      0      0      0
## 100288287      0      0      0      0      0      0
</code></pre>
</div>
<p>If you are interested in normalizing the gene counts to those from a given library size, say 40 million, you can use code like this to do so. You will need the total number of mapped reads. If you don't have that number, you can either calculate the sum of the coverage by sample and divide it by the read length or use the gene counts sum (by sample).</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## You might also want to normalize to a given library size, say 40 million
# here is a proxy for total mapped reads, although you might
# have that data somewhere else
totalMapped <- colSums(counts_genes)

norm_counts <- round(counts_genes / totalMapped * 4e7, 0)
head(norm_counts[, 1:6])</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##           HSB113 HSB123 HSB126 HSB130 HSB135 HSB136
## 100129027      0      0      0      0      0      0
## 100131902      0      0      0      0      0      0
## 100132288      0  37480  84623  47443  98384  26445
## 100133286  24021  61324  87217 129714  39024 114473
## 100151643      0      0      0      0      0      0
## 100288287      0      0      0      0      0      0
</code></pre>
</div>
<h1>Misc</h1>
<p>Other approaches for creating a gene-level count matrix from BigWig files could involve using <code>GenomicAlignments::summarizeOverlaps()</code> <a id='cite-GenomicAlignments'></a>(<a href='http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118'>Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>) although the <em>score</em> column created by <code>rtracklayer::import.bw()</code> gets ignored and you'll have to deal with it.</p>
<p>You could also use <em>genomic state</em> objects from <code>derfinder</code> <a id='cite-derfinder'></a>(<a href='http://www.biorxiv.org/content/early/2015/02/19/015370.abstract'>Collado-Torres, Frazee, Love, Irizarry, et al., 2015</a>). However, you have to deal with the complexity of exon overlaps and some missing ids as shown below.</p>
<div class="row">
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">library('derfinder')</code></pre>
<button class="warning R toggle btn btn-xs btn-warning">
<span class="glyphicon glyphicon-chevron-up"></span> R warning
</button>
<pre style="display:none"><code class="warning r">## Warning: replacing previous import by 'scales::alpha' when loading 'Hmisc'
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Check makeGenomicState()
## Exons from chr21 (hg19) included in the package
genomicState$fullGenome[ genomicState$fullGenome$theRegion == 'exon' ]</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## GRanges object with 2658 ranges and 4 metadata columns:
##        seqnames               ranges strand   |   theRegion
##           <Rle>            <IRanges>  <Rle>   | <character>
##      1    chr21   [9690071, 9690100]      +   |        exon
##      2    chr21   [9692178, 9692207]      +   |        exon
##      3    chr21   [9711935, 9712038]      +   |        exon
##      5    chr21   [9746121, 9746151]      +   |        exon
##      7    chr21   [9748830, 9748899]      +   |        exon
##    ...      ...                  ...    ... ...         ...
##   4985    chr21 [47740937, 47744424]      -   |        exon
##   4986    chr21 [48018531, 48019416]      -   |        exon
##   4988    chr21 [48020195, 48020288]      -   |        exon
##   4990    chr21 [48022191, 48022329]      -   |        exon
##   4992    chr21 [48024926, 48025035]      -   |        exon
##                        tx_id                              tx_name
##                <IntegerList>                      <CharacterList>
##      1                 72512                           uc002zkg.1
##      2                 72513                           uc021wgt.1
##      3                 72514                           uc011abu.2
##      5                 72514                           uc011abu.2
##      7                 72514                           uc011abu.2
##    ...                   ...                                  ...
##   4985 73460,73461,73462,... uc002zjf.3,uc002zjg.1,uc010gqj.2,...
##   4986           73464,73465                uc002zju.1,uc002zjv.1
##   4988                 73465                           uc002zjv.1
##   4990           73464,73465                uc002zju.1,uc002zjv.1
##   4992           73464,73465                uc002zju.1,uc002zjv.1
##                 gene
##        <IntegerList>
##      1              
##      2              
##      3              
##      5              
##      7              
##    ...           ...
##   4985           194
##   4986           227
##   4988           227
##   4990           227
##   4992           227
##   -------
##   seqinfo: 1 sequence from hg19 genome
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Load hg19 full file
load("/Users/lcollado/Dropbox/JHSPH/Code/derSoftware/GenomicState.Hsapiens.UCSC.hg19.knownGene.rda")
## Some are NA, some have more than 1 gene id
GenomicState.Hsapiens.UCSC.hg19.knownGene$codingGenome[ GenomicState.Hsapiens.UCSC.hg19.knownGene$codingGenome$theRegion == 'exon' ]$gene</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## CharacterList of length 217253
## [[1]] 100287102
## [[2]] 100287102
## [[3]] 100287102
## [[4]] 79501
## [[5]] <NA>
## [[6]] <NA>
## [[7]] 100133331 100132062 <NA>
## [[8]] 729759
## [[9]] <NA>
## [[10]] <NA>
## ...
## <217243 more elements>
</code></pre>
<button class="source R toggle btn btn-xs btn-primary">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</button>
<pre style=""><code class="source r">## Complicated case
GenomicState.Hsapiens.UCSC.hg19.knownGene$fullGenome[ GenomicState.Hsapiens.UCSC.hg19.knownGene$fullGenome$theRegion == 'exon' ][10]</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">## GRanges object with 1 range and 5 metadata columns:
##      seqnames           ranges strand |   theRegion         tx_id
##         <Rle>        <IRanges>  <Rle> | <character> <IntegerList>
##   15     chr1 [324439, 328581]      + |        exon     9,7,8,...
##                                   tx_name                   gene
##                           <CharacterList>        <CharacterList>
##   15 uc021oeh.1,uc009vjk.2,uc001aau.3,... 100133331,100132062,NA
##                            symbol
##                   <CharacterList>
##   15 LOC100133331,LOC100132062,NA
##   -------
##   seqinfo: 24 sequences from hg19 genome
</code></pre>
</div>
<h1>References</h1>
<p>Citations made with <code>knitcitations</code> <a id='cite-knitcitations'></a>(<a href='http://CRAN.R-project.org/package=knitcitations'>Boettiger, 2015</a>).</p>
<div class="row">
<button class="warning R toggle btn btn-xs btn-warning">
<span class="glyphicon glyphicon-chevron-up"></span> R warning
</button>
<pre style="display:none"><code class="warning r">## Warning in parse_Rd(Rd, encoding = encoding, fragment = fragment, ...): <connection>:7: unexpected END_OF_INPUT '
## '
</code></pre>
<p>
<a id='bib-knitcitations'></a><a href="#cite-knitcitations">[1]</a><cite> C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>. R package version 1.0.6. 2015. URL: <a href="http://CRAN.R-project.org/package=knitcitations">http://CRAN.R-project.org/package=knitcitations</a>.</cite>
</p>
<p>
<a id='bib-derfinder'></a><a href="#cite-derfinder">[2]</a><cite> L. Collado-Torres, A. C. Frazee, M. I. Love, R. A. Irizarry, et al. “derfinder: Software for annotation-agnostic RNA-seq differential expression analysis”. In: <em>bioRxiv</em> (2015). DOI: <a href="http://dx.doi.org/10.1101/015370">10.1101/015370</a>. URL: <a href="http://www.biorxiv.org/content/early/2015/02/19/015370.abstract">http://www.biorxiv.org/content/early/2015/02/19/015370.abstract</a>.</cite>
</p>
<p>
<a id='bib-derfinderData'></a><a href="#cite-derfinderData">[3]</a><cite> L. Collado-Torres, A. Jaffe and J. Leek. <em>derfinderData: Processed BigWigs from BrainSpan for examples</em>. R package version 0.103.1. 2015. URL: <a href="https://github.com/lcolladotor/derfinderData">https://github.com/lcolladotor/derfinderData</a>.</cite>
</p>
<p>
<a id='bib-rtracklayer'></a><a href="#cite-rtracklayer">[4]</a><cite> M. Lawrence, R. Gentleman and V. Carey. “rtracklayer: an R package for interfacing with genome browsers”. In: <em>Bioinformatics</em> 25 (2009), pp. 1841-1842. DOI: <a href="http://dx.doi.org/10.1093/bioinformatics/btp328">10.1093/bioinformatics/btp328</a>. URL: <a href="http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract">http://bioinformatics.oxfordjournals.org/content/25/14/1841.abstract</a>.</cite>
</p>
<p>
<a id='bib-GenomicAlignments'></a><a href="#cite-GenomicAlignments">[5]</a><cite> M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: <a href="http://dx.doi.org/10.1371/journal.pcbi.1003118">10.1371/journal.pcbi.1003118</a>. URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118}.">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118}.</a>
</p>
</cite>
</p>
</div>
<p><code>R</code> session information</p>
<div class="row">
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-down"></span> R message
</button>
<pre style=""><code class="message r">## Session info -----------------------------------------------------------------------------------------------------------
</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##  setting  value                                      
##  version  R version 3.2.0 Patched (2015-05-18 r68382)
##  system   x86_64, darwin10.8.0                       
##  ui       X11                                        
##  language (EN)                                       
##  collate  en_US.UTF-8                                
##  tz       America/New_York
</code></pre>
<button class="message R toggle btn btn-xs btn-info">
<span class="glyphicon glyphicon-chevron-down"></span> R message
</button>
<pre style=""><code class="message r">## Packages ---------------------------------------------------------------------------------------------------------------
</code></pre>
<button class="output R toggle btn btn-xs btn-success">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</button>
<pre style=""><code class="output r">##  package                           * version     date       source                                   
##  acepack                             1.3-3.3     2014-11-24 CRAN (R 3.2.0)                           
##  AnnotationDbi                     * 1.31.17     2015-06-18 Bioconductor                             
##  bibtex                              0.4.0       2014-12-31 CRAN (R 3.2.0)                           
##  Biobase                           * 2.29.1      2015-05-03 Bioconductor                             
##  BiocGenerics                      * 0.15.3      2015-06-30 Bioconductor                             
##  BiocParallel                        1.3.34      2015-07-10 Bioconductor                             
##  biomaRt                             2.25.1      2015-04-21 Bioconductor                             
##  Biostrings                          2.37.2      2015-05-07 Bioconductor                             
##  bitops                              1.0-6       2013-08-17 CRAN (R 3.2.0)                           
##  bumphunter                          1.9.1       2015-05-05 Bioconductor                             
##  cluster                             2.0.2       2015-06-19 CRAN (R 3.2.1)                           
##  codetools                           0.2-11      2015-03-10 CRAN (R 3.2.0)                           
##  colorspace                          1.2-6       2015-03-11 CRAN (R 3.2.0)                           
##  curl                                0.9.1       2015-07-04 CRAN (R 3.2.1)                           
##  DBI                                 0.3.1       2014-09-24 CRAN (R 3.2.0)                           
##  derfinder                         * 1.3.3       2015-06-15 Bioconductor                             
##  derfinderData                     * 0.103.1     2015-06-16 Bioconductor                             
##  derfinderHelper                     1.3.1       2015-06-20 Bioconductor                             
##  devtools                          * 1.8.0       2015-05-09 CRAN (R 3.2.0)                           
##  digest                              0.6.8       2014-12-31 CRAN (R 3.2.0)                           
##  doRNG                               1.6         2014-03-07 CRAN (R 3.2.0)                           
##  evaluate                            0.7         2015-04-21 CRAN (R 3.2.0)                           
##  foreach                             1.4.2       2014-04-11 CRAN (R 3.2.0)                           
##  foreign                             0.8-65      2015-07-02 CRAN (R 3.2.1)                           
##  formatR                             1.2         2015-04-21 CRAN (R 3.2.0)                           
##  Formula                             1.2-1       2015-04-07 CRAN (R 3.2.0)                           
##  futile.logger                       1.4.1       2015-04-20 CRAN (R 3.2.0)                           
##  futile.options                      1.0.0       2010-04-06 CRAN (R 3.2.0)                           
##  GenomeInfoDb                      * 1.5.8       2015-06-17 Bioconductor                             
##  GenomicAlignments                   1.5.11      2015-06-30 Bioconductor                             
##  GenomicFeatures                   * 1.21.13     2015-06-09 Bioconductor                             
##  GenomicFiles                        1.5.4       2015-07-03 Bioconductor                             
##  GenomicRanges                     * 1.21.16     2015-06-30 Bioconductor                             
##  ggplot2                             1.0.1.9000  2015-06-12 Github (hadley/ggplot2@3b6a126)          
##  git2r                               0.10.1.9000 2015-07-14 Github (ropensci/git2r@94831de)          
##  gridExtra                           0.9.1       2012-08-09 CRAN (R 3.2.0)                           
##  gtable                              0.1.2       2012-12-05 CRAN (R 3.2.0)                           
##  Hmisc                               3.16-0      2015-04-30 CRAN (R 3.2.0)                           
##  htmltools                           0.2.6       2014-09-08 CRAN (R 3.2.0)                           
##  httr                                1.0.0       2015-06-25 CRAN (R 3.2.1)                           
##  IRanges                           * 2.3.14      2015-07-03 Bioconductor                             
##  iterators                           1.0.7       2014-04-11 CRAN (R 3.2.0)                           
##  knitcitations                     * 1.0.6       2015-05-26 CRAN (R 3.2.0)                           
##  knitr                               1.10.5      2015-05-06 CRAN (R 3.2.0)                           
##  knitrBootstrap                      1.0.0       2015-06-09 Github (jimhester/knitrBootstrap@76c41f0)
##  lambda.r                            1.1.7       2015-03-20 CRAN (R 3.2.0)                           
##  lattice                             0.20-31     2015-03-30 CRAN (R 3.2.0)                           
##  latticeExtra                        0.6-26      2013-08-15 CRAN (R 3.2.0)                           
##  locfit                              1.5-9.1     2013-04-20 CRAN (R 3.2.0)                           
##  lubridate                           1.3.3       2013-12-31 CRAN (R 3.2.0)                           
##  magrittr                            1.5         2014-11-22 CRAN (R 3.2.0)                           
##  markdown                            0.7.7       2015-04-22 CRAN (R 3.2.0)                           
##  MASS                                7.3-42      2015-07-02 CRAN (R 3.2.1)                           
##  Matrix                              1.2-2       2015-07-08 CRAN (R 3.2.1)                           
##  matrixStats                         0.14.2      2015-06-24 CRAN (R 3.2.1)                           
##  memoise                             0.2.1       2014-04-22 CRAN (R 3.2.0)                           
##  munsell                             0.4.2       2013-07-11 CRAN (R 3.2.0)                           
##  nnet                                7.3-10      2015-06-29 CRAN (R 3.2.1)                           
##  pkgmaker                            0.22        2014-05-14 CRAN (R 3.2.0)                           
##  plyr                                1.8.3       2015-06-12 CRAN (R 3.2.0)                           
##  proto                               0.3-10      2012-12-22 CRAN (R 3.2.0)                           
##  qvalue                              2.1.0       2015-04-17 Bioconductor                             
##  R6                                  2.1.0       2015-07-04 CRAN (R 3.2.1)                           
##  RColorBrewer                        1.1-2       2014-12-07 CRAN (R 3.2.0)                           
##  Rcpp                                0.11.6      2015-05-01 CRAN (R 3.2.0)                           
##  RCurl                               1.95-4.7    2015-06-30 CRAN (R 3.2.1)                           
##  RefManageR                          0.8.63      2015-06-09 CRAN (R 3.2.0)                           
##  registry                            0.3         2015-07-08 CRAN (R 3.2.1)                           
##  reshape2                            1.4.1       2014-12-06 CRAN (R 3.2.0)                           
##  RJSONIO                             1.3-0       2014-07-28 CRAN (R 3.2.0)                           
##  rmarkdown                         * 0.7         2015-06-13 CRAN (R 3.2.0)                           
##  rngtools                            1.2.4       2014-03-06 CRAN (R 3.2.0)                           
##  rpart                               4.1-10      2015-06-29 CRAN (R 3.2.1)                           
##  Rsamtools                           1.21.13     2015-07-09 Bioconductor                             
##  RSQLite                             1.0.0       2014-10-25 CRAN (R 3.2.0)                           
##  rtracklayer                       * 1.29.12     2015-07-03 Bioconductor                             
##  rversions                           1.0.2       2015-07-13 CRAN (R 3.2.1)                           
##  S4Vectors                         * 0.7.10      2015-07-03 Bioconductor                             
##  scales                              0.2.5       2015-06-12 CRAN (R 3.2.0)                           
##  stringi                             0.5-5       2015-06-29 CRAN (R 3.2.1)                           
##  stringr                             1.0.0       2015-04-30 CRAN (R 3.2.0)                           
##  SummarizedExperiment                0.3.2       2015-07-03 Bioconductor                             
##  survival                            2.38-3      2015-07-02 CRAN (R 3.2.1)                           
##  TxDb.Hsapiens.UCSC.hg19.knownGene * 3.1.3       2015-05-19 Bioconductor                             
##  XML                                 3.98-1.3    2015-06-30 CRAN (R 3.2.0)                           
##  xml2                                0.1.1       2015-06-02 CRAN (R 3.2.0)                           
##  xtable                              1.7-4       2014-09-12 CRAN (R 3.2.0)                           
##  XVector                             0.9.1       2015-04-30 Bioconductor                             
##  yaml                                2.1.13      2014-06-12 CRAN (R 3.2.0)                           
##  zlibbioc                            1.15.0      2015-04-17 Bioconductor
</code></pre>
</div>
Date this protocol was last modified: 2015-07-22 13:41:19.
</div>
</div>
<div class="navbar navbar-fixed-bottom navbar-inverse">
<div class="container">
<div class="navbar-header">
<button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-responsive-collapse">
<span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span>
</button>
</div>
<div id="bottom-navbar" class="navbar-collapse collapse navbar-responsive-collapse">
<ul class="nav navbar-nav navbar-right">
<li class="nav">
<p class="navbar-text">
Toggle
</p>
</li>
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Code <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="dropdown-header">
Languages
</li>
<li class="active">
<a href="#" class="toggle-global source R" type="source.R">R</a>
</li>
<li>
<a href="#" type="all-source" class="toggle-global">All</a>
</li>
</ul>
</li>
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Output <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="dropdown-header">
Type
</li>
<li class="active">
<a href="#" class="toggle-global output" type="output">output</a>
</li>
<li class="active">
<a href="#" class="toggle-global message" type="message">message</a>
</li>
<li class="active">
<a href="#" class="toggle-global warning" type="warning">warning</a>
</li>
<li>
<a href="#" type="all-output" class="toggle-global">All</a>
</li>
</ul>
</li>
<li class="active">
<a href="#" type="figure" class="toggle-global">Figures</a>
</li>
</ul>
<ul class="nav navbar-nav navbar-left">
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Bootstrap theme <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="active">
<a class="theme-switch" title="Default" href="#">Default</a>
</li>
<li>
<a class="theme-switch" title="Amelia" href="#">Amelia</a>
</li>
<li>
<a class="theme-switch" title="Cerulean" href="#">Cerulean</a>
</li>
<li>
<a class="theme-switch" title="Cosmo" href="#">Cosmo</a>
</li>
<li>
<a class="theme-switch" title="Cyborg" href="#">Cyborg</a>
</li>
<li>
<a class="theme-switch" title="Journal" href="#">Journal</a>
</li>
<li>
<a class="theme-switch" title="Flatly" href="#">Flatly</a>
</li>
<li>
<a class="theme-switch" title="Readable" href="#">Readable</a>
</li>
<li>
<a class="theme-switch" title="Simplex" href="#">Simplex</a>
</li>
<li>
<a class="theme-switch" title="Slate" href="#">Slate</a>
</li>
<li>
<a class="theme-switch" title="Spacelab" href="#">Spacelab</a>
</li>
<li>
<a class="theme-switch" title="United" href="#">United</a>
</li>
</ul>
</li>
</ul>
<ul class="nav navbar-nav navbar-left">
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Code highlighting <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="active">
<a class="highlight-switch" title="HighlightJs" href="#">HighlightJs</a>
</li>
<li>
<a class="highlight-switch" title="Dark" href="#">Dark</a>
</li>
<li>
<a class="highlight-switch" title="Far" href="#">Far</a>
</li>
<li>
<a class="highlight-switch" title="Idea" href="#">Idea</a>
</li>
<li>
<a class="highlight-switch" title="Sunburst" href="#">Sunburst</a>
</li>
<li>
<a class="highlight-switch" title="Zenburn" href="#">Zenburn</a>
</li>
<li>
<a class="highlight-switch" title="Visual Studio" href="#">Visual Studio</a>
</li>
<li>
<a class="highlight-switch" title="Ascetic" href="#">Ascetic</a>
</li>
<li>
<a class="highlight-switch" title="Magula" href="#">Magula</a>
</li>
<li>
<a class="highlight-switch" title="Github" href="#">Github</a>
</li>
<li>
<a class="highlight-switch" title="Google Code" href="#">Google Code</a>
</li>
<li>
<a class="highlight-switch" title="Brown Paper" href="#">Brown Paper</a>
</li>
<li>
<a class="highlight-switch" title="School Book" href="#">School Book</a>
</li>
<li>
<a class="highlight-switch" title="IR Black" href="#">IR Black</a>
</li>
<li>
<a class="highlight-switch" title="Solarized - Dark" href="#">Solarized - Dark</a>
</li>
<li>
<a class="highlight-switch" title="Solarized - Light" href="#">Solarized - Light</a>
</li>
<li>
<a class="highlight-switch" title="Arta" href="#">Arta</a>
</li>
<li>
<a class="highlight-switch" title="Monokai" href="#">Monokai</a>
</li>
<li>
<a class="highlight-switch" title="Xcode" href="#">Xcode</a>
</li>
<li>
<a class="highlight-switch" title="Pojoaque" href="#">Pojoaque</a>
</li>
<li>
<a class="highlight-switch" title="Rainbow" href="#">Rainbow</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow" href="#">Tomorrow</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night" href="#">Tomorrow Night</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night Bright" href="#">Tomorrow Night Bright</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night Blue" href="#">Tomorrow Night Blue</a>
</li>
<li>
<a class="highlight-switch" title="Tomorrow Night Eighties" href="#">Tomorrow Night Eighties</a>
</li>
</ul>
</li>
</ul>
</div>
</div>
</div>
<link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" title="Default" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/amelia/bootstrap.min.css" title="Amelia" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/cerulean/bootstrap.min.css" title="Cerulean" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/cosmo/bootstrap.min.css" title="Cosmo" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/cyborg/bootstrap.min.css" title="Cyborg" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/journal/bootstrap.min.css" title="Journal" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/flatly/bootstrap.min.css" title="Flatly" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/readable/bootstrap.min.css" title="Readable" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/simplex/bootstrap.min.css" title="Simplex" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/slate/bootstrap.min.css" title="Slate" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/spacelab/bootstrap.min.css" title="Spacelab" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://netdna.bootstrapcdn.com/bootswatch/3.0.0/united/bootstrap.min.css" title="United" media="screen"></link><link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/default.min.css" title="HighlightJs" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/dark.min.css" title="Dark" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/far.min.css" title="Far" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/idea.min.css" title="Idea" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/sunburst.min.css" title="Sunburst" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/zenburn.min.css" title="Zenburn" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/vs.min.css" title="Visual Studio" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/ascetic.min.css" title="Ascetic" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/magula.min.css" title="Magula" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/github.min.css" title="Github" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/googlecode.min.css" title="Google Code" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/brown_paper.min.css" title="Brown Paper" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/school_book.min.css" title="School Book" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/ir_black.min.css" title="IR Black" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/solarized_dark.min.css" title="Solarized - Dark" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/solarized_light.min.css" title="Solarized - Light" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/arta.min.css" title="Arta" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/monokai.min.css" title="Monokai" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/xcode.min.css" title="Xcode" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/pojoaque.min.css" title="Pojoaque" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/rainbow.min.css" title="Rainbow" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow.min.css" title="Tomorrow" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night.min.css" title="Tomorrow Night" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night-bright.min.css" title="Tomorrow Night Bright" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night-blue.min.css" title="Tomorrow Night Blue" media="screen"></link> <link rel="alternate stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/tomorrow-night-eighties.min.css" title="Tomorrow Night Eighties" media="screen"></link>
</div>
<div id="push">

</div>
<div id="footer">
<div class="container">
<p class="text-muted" id="credit">
Styled with <a href="https://github.com/jimhester/knitrBootstrap">knitrBootstrap</a>
</p>
</div>
</div>
<link rel="stylesheet" id="theme" href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" media="screen"></link><link rel="stylesheet" id="highlight" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/default.min.css" media="screen"></link>
</div>
</body>
</html>
